### PR TITLE
Add upload_and_annotate command line tool

### PR DESCRIPTION
**Pull request by jvrakor**
@dblenkus @tristanbrown

### CODE REVIEW

File: resdk/scripts/upload_and_annotate.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+MOLECULE = {
+    'total RNA',
+    'polyA RNA',
+    'cytoplasmic RNA',
+    'nuclear RNA',
+    'genomic DNA',
+    'protein',
+    'other',
+}
+
```
[the remark this answers is not in the document]
Comment (jvrakor) [reply resdk/scripts/upload_and_annotate.py]: It is not restricted: https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/descriptors/reads.yml#L11. We may decide to restrict it in the validation script. Either to a list of options or at least that it shouldn't be left empty.

File: resdk/scripts/upload_and_annotate.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+            if diff1:
+                raise NameError(
+                    err_msg.format(', '.join(diff1), "are missing")
+                    )
+            if diff2:
+                raise NameError(
+                    err_msg.format(', '.join(diff2), "not recognised")
+                    )
+
+            for var_name, VAR in [('organism', ORGANISM), ('molecule', MOLECULE)]:
```
[the remark this answers is not in the document]
Comment (jvrakor) [reply resdk/scripts/upload_and_annotate.py]: See previous comment.

File: resdk/scripts/upload_and_annotate.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+        elif self.extension() in ['.txt', '.tab', '.tsv']:
+            self._read_text_file()
+        else:
+            raise TypeError(
+                "Annotation spreadsheet extension `{}` not recognised."
+                " Options are: `.xls`, `.xlsx`, `.xlsm`, `.txt`, "
+                "`.tab`, `.tsv`.".format(self.extension())
+            )
+
+    def validate(self):
```
[the remark this answers is not in the document]
Comment (jvrakor) [reply resdk/scripts/upload_and_annotate.py]: As said when we have a separate annotation script we might want to also check for ANNOTATOR and SOURCE. They are required to be present when saving a sample through the UI but not through ReSDK. The reason why we only check for MOLECULE and ORGANISM in the script is that they can't be empty strings as they are restricted. Other fields are allowed to be empty strings.

File: resdk/scripts/upload_and_annotate.py [hunk 4]
```diff
@@ -0,0 +1,340 @@
+"""Command line scripts."""
+# pylint: disable=logging-format-interpolation
+from __future__ import absolute_import, division, print_function, unicode_literals
```
[the remark this answers is not in the document]
Comment (jvrakor) [reply resdk/scripts/upload_and_annotate.py]: ReSDK should work in both python2 and python3.

File: resdk/scripts/upload_and_annotate.py [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+        reads.name = read_file.name
+
+        # Provide reads annotation
+        reads.descriptor_schema = 'reads'
+        reads.descriptor = read_file.reads_annotation
+        reads.save()
+
+        # Get sample object
+        reads.sample.delete(force=True)
+        main_sample = res.sample.create(name=read_file.name)
```
[the remark this answers is not in the document]
Comment (jvrakor) [reply resdk/scripts/upload_and_annotate.py]: This is a workaround so the slug is generated from the sample name. When uploading reads the sample slug and name are taken from the file name. We then change the name to the sample name defined in the annotation spreadsheet. We want to have a similar slug. Imagine a situation where there is a sample with the same name and slug already on the platform. Creating a new sample with the same name will result in this sample having the same name and almost the same slug (a suffix (an integer) will be added).

**Comment by jkokosar** [comment]
Closed in favour of https://github.com/genialis/resolwe-bio-py/pull/149

**Pull request closed without merging**